app/code/Magento/InventoryIndexer/Model/GetSourceEnabled.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Model\ResourceModel\Source;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+
+/**
+ * Get Enabled field value for Source by Source Code.
+ */
+class GetSourceEnabled
```

Comment:
Looks like we can merge classes in one and perform one query
We need this check only in one place